docs/rules/key-spacing.md

```diff
@@ -183,6 +184,110 @@ call({
 });
 ```
 
+### align
+
+The `align` option can take additional configuration through the `beforeColon`, `afterColon`, `mode`, and `on` options.
+
+If `align` is defined as an object, but not all of the parameters are provided, undefined parameters will default to the following:
+
+```js
+// Defaults
+align: {
+    "beforeColon": false,
+    "afterColon": true,
+    "on": "colon",
+    "mode": "strict"
+}
+```
+
+Examples of **correct** code for this rule with sample `{ "align": { } }` options:
+
+```js
+/*eslint key-spacing: ["error", {
+    "align": {
+        "beforeColon": true,
+        "afterColon": true,
+        "on": "colon"
+    }
+}]*/
+
+var obj = {
+    "one"   : 1,
+    "seven" : 7
+}
+```
+
+```js
+/*eslint key-spacing: ["error", {
+    "align": {
+        "beforeColon": false,
+        "afterColon": false,
+        "on": "value"
+    }
+}]*/
+
+var obj = {
+    "one":  1,
+    "seven":7
+}
+```
+
+### align and multiLine
+
+The `multiLine` and `align` options can differ, which allows for fine-tuned control over the `key-spacing` of your files.  `align` will **not** inherit from `multiLine` if `align` is configured as an object.
+
+`multiLine` will be used when a single key-value pair spans multiple lines.  For example, a function declaration or a nested object.  The `align` configuration is used when there is a group of properties in the the same object.  See below for an example of this distinction.
```

Comment:
> `multiLine` will be used when a single key-value pair spans multiple lines.

Not quite. `singleLine` and `multiLine` are about the object literals themselves, not the properties. For example, in the absence of any `align` options, all properties inside an object that has its opening and closing curly braces on the same line will obey the `singleLine` options, and all properties inside an object that has its opening and closing curly braces on different lines will obey the `multiLine` options:

```js
/*eslint key-spacing: ["error", {
    "singleLine": {
        "beforeColon": false,
        "afterColon": true
    },
    "multiLine": {
        "beforeColon": true,
        "afterColon": false
    }
}]*/

var singleLine = { key: value, key2: value2 }; // singleLine
var multiLine = {
    key :value, // multiLine
    key2 :value2 // multiLine
};
```

In the case of an object literal that spans multiple lines, `multiLine` will cover properties that are not part of a group. For properties that are part of a group, `align` will cover them if it is present, otherwise it will fall back to `multiLine`:

```js
/*eslint key-spacing: ["error", {
    "singleLine": {
        "beforeColon": false,
        "afterColon": true
    },
    "multiLine": {
        "beforeColon": true,
        "afterColon": false
    },
    "align": {
        "beforeColon": true,
        "afterColon": true
    }
}]*/

var singleLine = { "key": "value" };
var multiLine = {
    "key" :"value"
};
var combined = {
    key :value, // multiLine,

    key2 : value2, // align
    key3 : value3, // align

    key4 :value4 // multiLine
};
```